File: CPP/Code/Detail.cpp
#include <iostream>
#include "Produk.cpp"
using namespace std;

class Detail : public Produk{

	//atribut pakenya private
	private:
		int stok;
		int daya;
		double berat;

	//method yang bersifat public
	public:

		//constructor untuk inisiasi atribut
		Detail(string nama, string kode, string merk, int stok, int daya, double berat) : Produk(nama, kode, merk) {
            this->stok = stok;
            this->daya = daya;
            this->berat = berat;
		}

		//getter stok
		void setStok(int stok) {
			this->stok = stok;
		}
		
		//setter stok
		int getStok() {
			return this->stok;
		}
		
		//getter daya
		void setDaya(int daya) {
			this->daya = daya;
		}
		
		//setter daya
		int getDaya() {
			return this->daya;
		}
		
		//getter berat
		void setBerat(double berat) {
			this->berat = berat;
		}
		
		//setter berat
		double getBerat() {
			return this->berat;
		}
		
		~Detail(){ }

};

File: PHP/Code/Detail.php
<?php

require_once "Produk.php";

class Detail extends Produk {
    private int $stok;
    private int $daya;
    private float $berat;

    public function __construct(string $nama, string $kode, string $merk, int $stok, int $daya, float $berat) {
        parent::__construct($nama, $kode, $merk);
        $this->setStok($stok);
        $this->setDaya($daya);
        $this->setBerat($berat);
    }

    public function getStok(): int { 
        return $this->stok;
    }
    public function setStok(int $stok): void { 
        $this->stok = ($stok < 0) ? 0 : $stok; 
    }

    public function getDaya(): int { 
        return $this->daya;
    }
    public function setDaya(int $daya): void { 
        $this->daya = ($daya < 0) ? 0 : $daya; 
    }

    public function getBerat(): float { 
        return $this->berat;
    }
    public function setBerat(float $berat): void { 
        $this->berat = ($berat < 0) ? 0.0 : $berat; 
    }
}
